src/xdisplay.cpp

```diff
@@ -42,6 +46,63 @@ namespace xpyt
         }
     }
 
+    xeus::xjson display_pub_data(py::object obj)
+    {
+        py::module py_json = py::module::import("json");
+        xeus::xjson pub_data;
+
+        if (hasattr(obj, "_repr_mimebundle_"))
+        {
+            pub_data = xeus::xjson::parse(static_cast<std::string>(
+                py::str(py_json.attr("dumps")(obj.attr("_repr_mimebundle_")()))
+            ));
+        }
+        else if (hasattr(obj, "_repr_html_"))
+        {
+            pub_data["text/html"] = static_cast<std::string>(
+                py::str(obj.attr("_repr_html_")())
+            );
+        }
+        else if (hasattr(obj, "_repr_json_"))
+        {
+            pub_data["application/json"] = static_cast<std::string>(
+                py::str(obj.attr("_repr_json_")())
+            );
+        }
+        else if (hasattr(obj, "_repr_jpeg_"))
```

Comment:
Most of these are usually not else-if. All the available mimetype of an object get computed by IPython.  You usually don't know which mimetype a frontend can display.

Reply:
Ok, thanks!